### PR TITLE
A more basic retry queue

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- An attempt to simplify the queue from https://github.com/PostHog/posthog-js/pull/226
- Totally WIP. I haven't run the tests yet. 
- Putting it out there as an example of a simpler way to do the same thing. 

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/retry-queue.js [hunk 1]
```diff
@@ -6,41 +6,29 @@ export class RetryQueue extends RequestQueueScaffold {
...
-        this._pollerCounter = 1
-        this._areWeOnline = 'onLine' in window.navigator ? window.navigator.onLine : true
-        this._offlineBacklog = []
+        this.queue = []
+        this.areWeOnline = true
 
         if ('onLine' in window.navigator) {
+            this.areWeOnline = window.navigator.onLine
             window.addEventListener('online', () => {
-                this._handleWeAreNowOnline()
```
Comment (yakkomajuri) [inline src/retry-queue.js:17]: the reason for extracting this code was for test purposes where mocking online/offline event listeners would be messy and instead we can just call this

**Comment by yakkomajuri** [comment]
I'll take over this

File: src/retry-queue.js [hunk 2]
```diff
@@ -6,41 +6,29 @@ export class RetryQueue extends RequestQueueScaffold {
...
-        this._requestRetriesMap = {} // <RequestId, number>
-        this._counterToQueueMap = {}
-        this._pollerCounter = 1
-        this._areWeOnline = 'onLine' in window.navigator ? window.navigator.onLine : true
-        this._offlineBacklog = []
+        this.queue = []
+        this.areWeOnline = true
 
         if ('onLine' in window.navigator) {
+            this.areWeOnline = window.navigator.onLine
```
Comment (yakkomajuri) [inline src/retry-queue.js:13]: not all browsers support this, although I guess having it as `undefined` achieves the same purpose as `false`, despite a bit messier types

Comment (mariusandra) [reply src/retry-queue.js:13]: thanks for taking over and feel free to change anything... I never really ran it :D

**Comment by yakkomajuri** [comment]
This should now be done. Finished up the changes and updated all the tests. Hoping we can get this and #226 in @mariusandra

**Pull request merged**